tests/lib/rules/operator-assignment.js

```diff
@@ -214,6 +215,26 @@ ruleTester.run("operator-assignment", rule, {
         output: "foo = foo ** bar",
         options: ["never"],
         errors: UNEXPECTED_OPERATOR_ASSIGNMENT
+    }, {
+        code: "foo *= bar + 1",
+        output: "foo = foo * (bar + 1)",
+        options: ["never"],
+        errors: UNEXPECTED_OPERATOR_ASSIGNMENT
+    }, {
+        code: "foo -= bar - baz",
+        output: "foo = foo - (bar - baz)",
+        options: ["never"],
+        errors: UNEXPECTED_OPERATOR_ASSIGNMENT
+    }, {
+        code: "foo += bar = 1",
+        output: "foo = foo + (bar = 1)",
+        options: ["never"],
+        errors: UNEXPECTED_OPERATOR_ASSIGNMENT
+    }, {
+        code: "foo *= (bar + 1)",
```

Comment:
I added a [similar test](https://github.com/eslint/eslint/blob/8914fc80c11e7cf5dce131d0a499a530b206d6a8/tests/lib/rules/operator-assignment.js#L224-L227) for `foo -= bar - baz`, but I can add one for `+` as well.

Note that addition isn't associative in JS, so the parentheses on the right would still be necessary:

```js
foo = 1;
bar = 2;
baz = '3';

foo += bar + baz; // => foo is "123"
```

```js
foo = 1;
bar = 2;
baz = '3';
foo = foo + bar + baz; // => foo is "33"
```